src/BehaviorsSDKManaged/Microsoft.Xaml.Interactivity.Shared/Core/ChangeDependencyPropertyAction.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if WinUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Markup;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Markup;
+#endif
+
+namespace Microsoft.Xaml.Interactivity;
+
+/// <summary>
+/// An action that will change a specified dependency property to a specified value when invoked.
+/// </summary>
+[ContentProperty(Name = "Value")]
+public sealed class ChangeDependencyPropertyAction : DependencyObject, IAction
+{
+    /// <summary>
+    /// Identifies the <seealso cref="TargetObject"/> dependency property.
+    /// </summary>
+    public static readonly DependencyProperty TargetObjectProperty = DependencyProperty.Register(
+        "TargetObject",
+        typeof(DependencyObject),
+        typeof(ChangeDependencyPropertyAction),
+        new PropertyMetadata(null));
+
+    /// <summary>
+    /// Identifies the <seealso cref="Value"/> dependency property.
+    /// </summary>
+    public static readonly DependencyProperty ValueProperty = DependencyProperty.Register(
+        "Value",
+        typeof(object),
+        typeof(ChangeDependencyPropertyAction),
+        new PropertyMetadata(null));
+
+    /// <summary>
+    /// Gets or sets the dependency property to change. This is not a dependency property, due to framework restrictions.
```

Comment:
All dependency properties for this class break when one of the DPs has DependencyProperty itself as its type. Meta-DPs are reserved to the framework itself apparently...

Reply:
🤔 Interesting, I know sometimes not having it as a DP could cause VS to bubble up other warnings when binding to things, but since it'd only ever be a OneTime binding sort of scenario, I can't imagine it being an issue.